pages/blog/[uid].js

```diff
@@ -42,12 +45,12 @@ const Post = ({ post }) => {
   return null;
 };
 
-export async function getStaticProps({ params, preview = null, previewData = {} }) {
-  const { ref } = previewData
+export async function getStaticProps({ params, previewData }) {
+  const ref = previewData ? previewData.ref : null

```

Comment:
There's no need to name this `ref` here and change it to `previewRef` down below. You can just name it `previewRef` directly right here.

Reply:
The application throughs an error in the `getByUid` function when this is renamed `previewRef` here.

Reply:
You'll need to update the query options to this: `const options = previewRef ? { ref: previewRef } : null`